hashedindex/textparser.py

```diff
@@ -92,15 +99,13 @@ def word_tokenize(text, stopwords=_stopwords, ngrams=None, min_length=0, ignore_
     validate_stemmer(stemmer)
 
     text = re.sub(re.compile('\'s'), '', text)  # Simple heuristic
-    text = re.sub(_re_punctuation, '', text)
+    text = text if retain_punctuation else re.sub(_re_punctuation, '', text)
     text = text if retain_casing else text.lower()
 
     matched_tokens = re.findall(_re_token, text)
     for tokens in get_ngrams(matched_tokens, ngrams):
         for i in range(len(tokens)):
-            tokens[i] = tokens[i].strip(punctuation)
```

Comment:
:hourglass_flowing_sand: I think this line should be restored.  I'll add some before/after test coverage to illustrate why.

Reply:
:heavy_check_mark: Good news - no need for additional coverage here; this case was already [covered by a test](https://github.com/MichaelAquilina/hashedindex/blob/783c7adfaf11e73fe40bc3535b36f029cc28b8b2/tests/test_parser.py#L105-L108) prior to the commit this changeset branches from.  The test name could possibly be updated; PR #19 is available to optionally make that change if it seems worthwhile.